modules/tinymce/src/themes/silver/main/ts/ui/dialog/IFrame.ts

```diff
@@ -81,8 +81,11 @@ const getDynamicSource = (initialData: Optional<string>, stream: boolean): IFram
       });
   };
 
-  // TINY-10097: Throttle to reduce flickering, as the document.write() method still observes significant flickering on Safari.
-  const writeValueThrottler = isSafari ? Optional.some(Throttler.first(writeValue, 500)) : Optional.none();
+  // TINY-10078: On Firefox, throttle to 200ms allow improve scrolling experience. Since we are manually maintaining previous scroll
+  // position on each update, when updating too rapidly, attempting to scroll around the iframe can feel stuck.
+  // TINY-10097: On Safari, throttle to 500ms reduce flickering as the document.write() method still observes significant flickering.
+  // Also improves scrolling, as scroll positions are maintained manually similar to Firefox.
+  const writeValueThrottler = isSafariOrFirefox ? Optional.some(Throttler.first(writeValue, isSafari ? 500 : 200)) : Optional.none();
```

Comment:
Looking at this in context I'd like to simplify the code a bit.

For starters, perhaps `isSafariOrFirefox` could be an optional number. Then the `500` and `200` values are defined along with other impactful constants at the top, and this becomes a much nicer `Optional.map`.

Ideally I'd like to avoid `writeValueThrottler.fold` below, but I'm not sure how we do that while retaining the single `Throttler` instance.

Reply:
That's a great suggestion, thanks 👍

Reply:
I actually meant replacing the `isSafariOrFirefox` boolean with the optional number, but that works 😂